pkg/apis/extensions/v1beta1/types.go

```diff
@@ -342,6 +342,9 @@ type DeploymentStatus struct {
 
 	// Total number of unavailable pods targeted by this deployment.
 	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty"`
+
+	// Total number of pods a deployment should have at any given point.
+	EnactedReplicas int32 `json:"enactedReplicas,omitempty"`
```

Comment:
Got it working so I thought of pushing it. I have been testing it locally and it seems to work fine (d.spec.replicas in the case of one rs, d.spec.replicas+maxSurge in the case of multiple rs during a rolling deployment). I can add more e2e testing around it if I finally resolve why my e2e test fails at the start..
```
Feb 25 09:26:12.267: INFO: Verifying that the required pods have come up...
Feb 25 09:26:12.334: INFO: Pod name redis: Found 0 pods out of 1
Feb 25 09:26:17.365: INFO: Pod name redis: Found 1 pods out of 1
STEP: ensuring each pod is running
Feb 25 09:26:17.365: INFO: Waiting up to 5m0s for pod redis-set-8pife status to be running
Feb 25 09:26:17.406: INFO: Found pod 'redis-set-8pife' on node 'e2e-gce-master-0-minion-wesx'
STEP: trying to dial each unique pod
Feb 25 09:26:17.574: INFO: Controller redis: Failed to GET from replica 1 [redis-set-8pife]: an error on the server has prevented the request from succeeding (get pods redis-set-8pife):
```
The pod comes up but never responds
```
failed to wait for pods responding: timed out waiting for the condition
```
Got any clues? It's exactly the same way every other e2e test starts, the only difference is that I am using redis instead of nginx as an image...